tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -37,7 +37,14 @@ private static void LogMatchesIfDebugEnabled(string? result, bool blocked)
             for (var i = 0; i < results?.Length; i++)
             {
                 var match = results[i];
-                Log.Debug(blocked ? "DDAS-0012-02: Blocking current transaction (rule: {RuleId})" : "DDAS-0012-01: Detecting an attack from rule {RuleId}", match.Rule);
+                if (blocked)
```

Comment:
is it because it's more readable?

Reply:
This is a slightly annoying symptom of the Serilog analyzer - it sees the expression 

```csharp
blocked ? "DDAS-0012-02: Blocking current transaction (rule: {RuleId})" : "DDAS-0012-01: Detecting an attack from rule {RuleId}"
```

and knows that it's not a _constant_ string. What it can't tell is that it's one of two constant strings, so it's fine in reality. We could have validly suppressed the warning here, but my concern is we would change it to something that's _actually_ not constant, and not get the warning. Switching to the new approach is "safe" at the expense of annoying extra verbosity 🤷